media/js/privacy/firefox-privacy.js

```diff
@@ -0,0 +1,288 @@
+                return propertyValue ? propertyValue.toString() : propertyValue;
+            } else {
+                return data.elem.dataset[formatAsDatasetProperty(data.property)];
+            }
+        }
+    }
+
+    /**
+     * Takes a string in the format `data-tabpanel-open-text` and
+     * returns a string formatted such that it can be used as a
+     * property accessor on a `dataset`
+     * Example:
+     * input: data-tabpanel-open-text
+     * returns tabpanelOpenText
+     *
+     * @param {String} property - The property to convert as a string
+     * @returns the formatted string
+     */
+    function formatAsDatasetProperty(property) {
+        var formattedProperty = '';
+        // first strip of the `data-`
+        var stripped = property.substr(property.indexOf('-') + 1);
+        // split the result into an array
+        var split = stripped.split('-');
+
+        /* start by adding the first item in the array
+        to the `formattedProperty` which will be returned */
+        formattedProperty = split[0];
+
+        // if the resulting array has more than one item
+        if (split.length > 1) {
+            // loop over the array, starting at the second item
+            for (var i = 1, l = split.length; i < l; i++) {
+                var currentString = split[i];
+                formattedProperty += currentString.replace(
+                    currentString.substr(0,1),
+                    currentString.substr(0,1).toUpperCase()
+                );
+            }
+        }
+        return formattedProperty;
+    }
+
+    /**
+     * Hides *all* topics across sections
+     */
+    function hideAllTopicContent() {
+        for (var i = 0, l = topics.length; i < l; i++) {
+            var divElem = topics[i].querySelector('div');
+            var topicHeading = topics[i].querySelector('h3');
+
+            divElem.classList.add('hidden');
+            divElem.setAttribute('aria-hidden', true);
+
+            topicHeading.classList.add('collapsed');
+        }
+    }
+
+    /**
+     * Runs a set of functions on page load.
+     */
+    function initPage() {
+        collectAllTopics();
+        hideAllTopicContent();
+        addMasterToggles();
+        showInitialTopic();
+    }
+
+    /**
+     * On load, expands the first topic of the first main section. This
+     * also calls `addDataChoicesWidget` if on Fx desktop.
+     */
+    function showInitialTopic() {
+        var initialTopic = topicHeaders[0].querySelector('section');
+        var initialTopicContent = initialTopic.querySelector('div');
+        var initialTopicHeading = initialTopic.querySelector('h3');
+
+        initialTopicContent.classList.remove('hidden');
+        initialTopicContent.setAttribute('aria-hidden', false);
+
+        initialTopicHeading.classList.remove('collapsed');
+
+        if (Mozilla.Client.isFirefoxDesktop) {
+            addDataChoicesWidget(initialTopicContent);
```

Comment:
Would it be worth adding this behind a call to `Mozilla.UITour.ping()`? That was the button will only show if UITour is working.

Reply:
Oh, that is even better. Thanks!

Reply:
I'd still keep the check for `Mozilla.Client.isFirefoxDesktop` fwiw (to avoid calling UITour entirely for other browser)